src/services/language.js

```diff
@@ -0,0 +1,162 @@
+import importedLibs from 'lang';
+import Kleio from 'kleio';
+import store from 'flux/store';
+import deepExtend from 'deep-extend';
+
+/**
+ * Service that manages imported language libs. Based on language code
+ */
+class LanguageService {
+  constructor(language = LanguageService.defaultLanguage) {
+    const kleio = new Kleio(null, Kleio.ENV_MODES.DEV);
+
+    this._record = kleio.record.bind(kleio);
+
+    this._listeners = [];
+
+    this.language = language;
+
+    this._setLanguageFromStore();
+    store.subscribe(this._setLanguageFromStore);
+  }
+
+  static defaultLanguage = 'en';
+
+  /**
+   * Returns the current country code.
+   *
+   * @return {String} country code
+   */
+  get language() {
+    return this._languageCode;
+  }
+
+  /**
+   * Set the current code and update the lodaded language file.
+   *
+   * @param  {String} value country code specification, ex: en
+   */
+  set language(value = LanguageService.defaultLanguage) {
+    if (typeof value !== 'string') {
+      const error = new Error('Passed language value must be of type string.');
+      this._record(
+        'LanguageService',
+        'Invalid type was passed to set language',
+        Kleio.levels.INFO,
+        error.stack, {
+          value
+        }
+      );
+
+      throw error;
+    }
+
+    this._languageCode = value;
+
+    this._setLanguageLibrary();
+    this._triggerListeners();
+  }
+
+  _setLanguageFromStore = () => {
+    const appState = store.getState().app;
+
+    if (appState.language && appState.language !== this.language) {
+      this.language = appState.language;
+    }
+  };
+
+  /**
+   * _getLanguageCodeHierarchy Create array of language codes based on current language code..
+   * Example: If  current language code is 'sv_SE', return value would be `['en', 'sv', 'sv_SE]`
+   * @return {array} Array of language codes
+   */
+  _getLanguageCodeHierarchy() {
+    return (this._languageCode ? this._languageCode : LanguageService.defaultLanguage)
+      .split('_')
+      .reduce((prevValue, currentValue, currentIndex) => {
+        const newValue = (currentIndex === 0) ? currentValue : `${prevValue[prevValue.length - 1]}_${currentValue}`;
+
+        if (prevValue.indexOf(newValue) === -1) {
+          return prevValue.concat(newValue);
+        }
+
+        return prevValue;
+      }, [LanguageService.defaultLanguage]);
+  }
+
+  /**
+   * Creates a language object based on the currently defined language code.
+   */
+  _setLanguageLibrary() {
+    const translations = {},
+      languageCodes = this._getLanguageCodeHierarchy();
+
+    languageCodes.forEach(languageCode => {
+      deepExtend(translations, importedLibs[languageCode] || {});
+    });
+
+    this._currentLibrary = translations;
+  }
+
+  /**
+   * Registration method to hook on lanugage listener. Automatically unsubscribe on unmount.
+   *
+   * @param  {ReactComponent} context component context.
+   */
+  use(context) {
```

Comment:
I think we should rethink this technique later and probably use the [HOC](https://github.com/krasimir/react-in-patterns/tree/master/patterns/higher-order-components) pattern. To hook on to unmount in a more acceptable way?